src/argsparser.c

```diff
@@ -301,6 +302,20 @@ void parse_recurring_delay(char *arg)
     free(x);
 }
 
+int isnumber(char *pin)
```

Comment:
The code section at line 332 can be executed when the specified argument pin contains 8, 7 or 4 digits. If there is only 4 digits, it will set it as first part of pin, and continue bruteforcing second part. If there's 7 digits, it will calculate checksum digit and append it to the pin, entering one-shot mode. It also accepts 8 digit pin in two possible scenarios: with correct checksum and without it (-X "exhaustive" option is required for that).

So, any other pin values would set it into arbitrary string mode, that's why I added this function.

If we use `wps_pin_checksum()` here, then we can't set 4 or 7 pin anymore (they will be interpreted as arbitrary strings).